### PR TITLE
[mt] Support learning rate.

### PR DESCRIPTION
**Pull request by trivialfis**
- Use the learning rate for leaf weights.
- Hide the MT implementation behind the debug flag.

ref https://github.com/dmlc/xgboost/issues/9043

### CODE REVIEW

**Comment by trivialfis** [comment]
cc @rongou
